tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/views/execution_data/execution_data_container.ts

```diff
@@ -94,5 +102,47 @@ export class ExecutionDataContainer {
     )
   );
 
+  readonly debugTensorDtypes$ = this.store.pipe(
+    select(
+      createSelector(
+        getFocusedExecutionData,
+        getDtypesMap,
+        (
+          execution: Execution | null,
+          dtypesMap: DtypesMap
+        ): string[] | null => {
+          if (execution === null || execution.debug_tensor_values === null) {
+            return null;
+          }
+          if (
+            execution.tensor_debug_mode !== TensorDebugMode.FULL_HEALTH &&
+            execution.tensor_debug_mode !== TensorDebugMode.SHAPE
+          ) {
+            // TODO(cais): Add logic for other TensorDebugModes with dtype info.
+            return null;
+          }
+          const dtypes: string[] = [];
+          for (const tensorValue of execution.debug_tensor_values) {
+            if (tensorValue === null) {
+              dtypes.push(UNKNOWN_DTYPE_NAME);
+            } else {
+              const dtypeEnum = String(
+                execution.tensor_debug_mode === TensorDebugMode.FULL_HEALTH
+                  ? tensorValue[2] // tensor_debug_mode: FULL_HEALTH
+                  : tensorValue[1] // tensor_debug_mode: SHAPE
+              );
+              dtypes.push(
+                dtypeEnum in dtypesMap
+                  ? dtypesMap[dtypeEnum].name
+                  : UNKNOWN_DTYPE_NAME
+              );
+            }
+          }
```

Comment:
consider using `map` instead since it is more semantically meaningful than `for-of` and `push`.

Reply:
No longer relevant now we're using the genrule.